packages/astro/src/core/compile/style.ts

```diff
@@ -11,18 +12,21 @@ export function createStylePreprocessor({
 }: {
 	filename: string;
 	viteConfig: ResolvedConfig;
-	cssDeps: Set<string>;
+	cssDeps: CompileCssResult['dependencies'][];
 	cssTransformErrors: Error[];
 }): TransformOptions['preprocessStyle'] {
+	let processedStylesCount = 0;
+
 	return async (content, attrs) => {
+		const index = processedStylesCount++;
 		const lang = `.${attrs?.lang || 'css'}`.toLowerCase();
-		const id = `${filename}?astro&type=style&lang${lang}`;
+		const id = `${filename}?astro&type=style&index=${index}&lang${lang}`;
 		try {
 			const result = await preprocessCSS(content, id, viteConfig);
 
-			result.deps?.forEach((dep) => {
-				cssDeps.add(dep);
-			});
+			if (result.deps) {
+				cssDeps[index] = [...result.deps].map((dep) => normalizePath(dep));
```

Comment:
```suggestion
				cssDeps[index] = result.deps.map((dep) => normalizePath(dep));
```

Not sure the spread operator is needed..

Reply:
`result.deps` is a `Set` and doesn't have `.map`